File: 1351. Count Negative Numbers in a Sorted Matrix/main.py
from typing import List


class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        count = 0

        for line in grid:
            left, right = 0, len(line) - 1

            while left <= right:
                mid = left + (right - left) // 2

                if line[mid] >= 0:
                    left = mid + 1
                else:
                    right = mid - 1

            count += len(line) - left

        return count

File: 1351. Count Negative Numbers in a Sorted Matrix/main.c
#include <stdbool.h>
#include <stdio.h>

int countNegatives(int **grid, int gridSize, int *gridColSize) {
    int count = 0;

    for (int i = 0; i < gridSize; i++) {
        int left = 0;
        int right = gridColSize[i] - 1;

        while (left <= right) {
            int mid = left + (right - left) / 2;

            if (grid[i][mid] >= 0) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }

        count += gridColSize[i] - left;
    }

    return count;
}